### PR TITLE
Add specs for @forward ... with

### PR DESCRIPTION
**Pull request by nex3**
See sass/sass#2744

[skip dart-sass]

### CODE REVIEW

File: spec/directives/forward/error/syntax.hrx [hunk 1]
```diff
@@ -347,3 +347,133 @@ Error: @forward rules must be written before any other rules.
...
+  ,
+1 | @forward "other" with;
+  |                      ^
+  '
+  input.scss 1:22  root stylesheet
+
+<===>
+================================================================================
+<===> with/before_as/input.scss
+@forward "other" with ($a: b) as c-*;
```
[the remark this answers is not in the document]
Comment (nex3) [reply spec/directives/forward/error/syntax.hrx]: https://github.com/sass/sass-spec/blob/0749da435328e48acf2a54f0299b4d78cefc8d63/spec/directives/use/error/syntax.hrx#L515-L525

:smile:

[the remark this answers is not in the document]
Comment (mirisuzanne) [reply spec/directives/forward/error/syntax.hrx]: Ha - I guess I was referring to user-facing documentation that these phrases have a set order :)

[the remark this answers is not in the document]
Comment (nex3) [reply spec/directives/forward/error/syntax.hrx]: Generally, user-facing documentation is not 100% thorough—we gloss over some details that are unlikely to matter in order to keep it readable and beginner-friendly. The prose specification is where you should look for the most thorough and rigorous documentation, and it's also [clear about this](https://github.com/sass/sass/blob/master/spec/at-rules/use.md#syntax).

**Comment by nex3** [comment]
> It would also be good to add some specs for the interaction between `@forward ... with` and `@import`.

Done.

**Pull request merged**